src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -1092,11 +1072,7 @@ namespace OpenLoco::Vehicles
             auto invertedDirection = sprite_yaw ^ (1 << 5);
             auto xyFactor = Math::Trigonometry::computeXYVector(positionFactor, invertedDirection) / 2;
 
-            Map::Pos3 loc = {
-                static_cast<int16_t>(x + xyFactor.x),
-                static_cast<int16_t>(y + xyFactor.y),
-                static_cast<int16_t>(z + vehicleObject->animation[num].height)
-            };
+            Map::Pos3 loc = position + Map::Pos3{ static_cast<coord_t>(xyFactor.x), static_cast<coord_t>(xyFactor.y), vehicleObject->animation[num].height };
```

Comment:
Can we avoid these casts as well, by using the following?
```suggestion
            Map::Pos3 loc = position + Map::Pos3(xyFactor.x, xyFactor.y, vehicleObject->animation[num].height);
```